src/draw/DefaultDrawer.js

```diff
@@ -48,25 +53,97 @@ class DefaultDrawer {
         .node().innerHTML = icon;
     });
 
-    this.drawDefaultResource(component);
+    this.initializeComponents(components, svgContainer);
+    this.drawDefaultModel(svgContainer);
 
     componentsSelection.exit()
       .remove();
     return true;
   }
 
+  /**
+   * Initialize all draw option of the components.
+   * @param {Component[]} components - List of Components.
+   * @param {Object} svgContainer - D3 Object contains components SVGs.
+   */
+  initializeComponents(components, svgContainer) {
+    const componentsToInit = [];
+    const componentMaxSize = { width: 0, height: 0 };
+
+    svgContainer.each((_data, index, array) => {
+      const component = components[index];
+      if (!component.drawOption) {
+        component.drawOption = new ComponentDrawOption();
+        componentsToInit.push(components[index]);
+      }
+      this.storeComponentSize(array[index], component);
+      if (component.drawOption.width > componentMaxSize.width) {
+        componentMaxSize.width = component.drawOption.width;
+      }
+      if (component.drawOption.height > componentMaxSize.height) {
+        componentMaxSize.height = component.drawOption.height;
+      }
+    });
+
+    if (componentsToInit.length > 0) {
+      const viewportWidth = document.querySelector('#viewport').clientWidth;
+      componentsToInit.forEach((component, index) => this.setComponentPosition(
+        component,
+        index,
+        viewportWidth,
+        componentMaxSize,
+      ));
+    }
+  }
+
+  /**
+   * Defines the position of the components without drawOption by computing
```

Comment:
remove `without drawOption`, because in future we can call these function to reset all components position :)